Mathlib/Topology/VectorBundle/Riemannian.lean

```diff
@@ -0,0 +1,241 @@
+/-
+Copyright (c) 2025 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.InnerProductSpace.LinearMap
+import Mathlib.Topology.VectorBundle.Constructions
+import Mathlib.Topology.VectorBundle.Hom
+
+/-! # Riemannian vector bundles
+
+Given a vector bundle over a manifold whose fibers are all endowed with a scalar product, we
+say that this bundle is Riemannian if the scalar product depends continuously on the base point.
+
+We introduce a typeclass `[IsContinuousRiemannianBundle F E]` registering this property.
+Under this assumption, we show that the scalar product of two continuous maps into the same fibers
+of the bundle is a continuous function.
+
+If one wants to endow an existing vector bundle with a Riemannian metric, there is a subtlety:
+the inner product space structure on the fibers should give rise to a topology on the fibers
+which is defeq to the original one, to avoid diamonds. To do this, we introduce a
+class `[RiemannianBundle E]` containing the data of a scalar
+product on the fibers defining the same topology as the original one. Given this class, we can
+construct `NormedAddCommGroup` and `InnerProductSpace` instances on the fibers, compatible in a
+defeq way with the initial topology. If the data used to register the instance `RiemannianBundle E`
+depends continuously on the base point, we register automatically an instance of
+`[IsContinuousRiemannianBundle F E]` (and similarly if the data is smooth).
+
+The general theory should be built assuming `[IsContinuousRiemannianBundle F E]`, while the
+`[RiemannianBundle E]` mechanism is only to build data in specific situations.
+-/
+
+open Bundle ContinuousLinearMap ENat
+open scoped Topology
+
+variable
+  {B : Type*} [TopologicalSpace B]
+  {F : Type*} [NormedAddCommGroup F] [NormedSpace ℝ F]
+  {E : B → Type*} [TopologicalSpace (TotalSpace F E)] [∀ x, NormedAddCommGroup (E x)]
+  [∀ x, InnerProductSpace ℝ (E x)]
+  [FiberBundle F E] [VectorBundle ℝ F E]
+
+local notation "⟪" x ", " y "⟫" => inner ℝ x y
+
+variable (F E) in
+/-- Consider a real vector bundle in which each fiber is endowed with a scalar product.
+We say that the bundle is Riemannian if the scalar product depends continuously on the base point.
```

Comment:
What do you think about this? (If so, probably also in the module doc-string above.)
```suggestion
We say that the bundle is *Riemannian* if the scalar product depends continuously on the base point.